golang/lib.go

```diff
@@ -0,0 +1,306 @@
+package usearch
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+	"sync"
+	"unsafe"
+)
+
+/*
+#include <stdlib.h>
+#include "golang.h"
+// avoid some verbosity by using void* index here
+// proper typed pointer in the cpp file definitions
+void* new_index(char* metric_str, int metric_len,
```

Comment:
I would generally recommend adding a prefix to function names, like `usearch_` to avoid collisions with more common verbs, aka `save`, `size`, `capacity`...